control/ping.yaml

```diff
@@ -0,0 +1,93 @@
+openapi: 3.1.0
+
+info:
+  title: Ping models
+  description: >-
+    All ping schemas
+  version: ^0.0.0
+
+components:
+  schemas:
+    Ping:
+      type: object
+      properties:
+        choice:
+          description: ipv4 or ipv6 ping
+          type: string
+          enum: [ipv4, ipv6]
+          default: ipv4
+        ipv4:
+          $ref: '#/components/schemas/Ping.Ipv4'
+        ipv6:
+          $ref: '#/components/schemas/Ping.Ipv6'
+
+    Ping.Ipv4:
+      type: object
+      properties:
+        src_name:
+          description: >-
+            A base IPv4 interface
+          type: string
+          x-constraint:
+          - '/components/schemas/Device.Ipv4Base/properties/name'
+        dst_ips:
+          description: >-
+              Array of destination ipv4 addresses
+          type: array
+          items:
+            description: >-
+              destination address
+            type: string
+            format: ipv4
+    
+    Ping.Ipv6:
+      type: object
+      properties:
+        src_name:
+          description: >-
+            A base IPv6 interface
+          type: string
+          x-constraint:
+          - '/components/schemas/Device.Ipv6Base/properties/name'
+        dst_ips:
+          description: >-
+              Array of destination ipv6 addresses
+          type: array
+          items:
+            description: >-
+              destination address
+            type: string
+            format: ipv6
+    
+    Ping.Response:
+      description: >-
+        A container for the different ping responses.
+      type: object
+      properties:
+        responses:
+          type: array
+          items:
+            $ref: '#/components/schemas/Response'
+
+    Response:
+      type: object
+      properties:
+        src_name:
+          description: >-
+            The name of the source the ping is requested
```

Comment:
The name of the source IPv4 or IPv6 interface from which ping was sent.

Reply:
updated